front/lib/api/assistant/conversation.ts

```diff
@@ -630,16 +570,167 @@ export async function* retryAgentMessage(
   | AgentGenerationSuccessEvent
   | AgentMessageSuccessEvent
 > {
-  yield {
-    type: "agent_error",
-    created: Date.now(),
-    configurationId: "foo",
-    messageId: "bar",
-    error: {
-      code: "not_implemented",
-      message: "Not implemented",
+  const {
+    event,
+    agentMessageRow,
+  }: {
+    event: AgentMessageNewEvent | AgentErrorEvent;
+    agentMessageRow?: AgentMessage;
+  } = await front_sequelize.transaction(async (t) => {
+    const messageRow = await Message.findOne({
+      where: {
+        conversationId: conversation.id,
+        id: message.id,
+      },
+      include: [
+        {
+          model: AgentMessage,
+          as: "agentMessage",
+          required: true,
+        },
+      ],
+      transaction: t,
+    });
+
+    if (!messageRow) {
+      return {
+        event: {
+          type: "agent_error",
+          created: Date.now(),
+          configurationId: message.configuration.sId,
+          messageId: message.sId,
+          error: {
+            code: "not_found",
+            message: "Message row not found",
+          },
+        },
+      };
+    }
+    const agentMessageRow = messageRow.agentMessage;
+    if (!agentMessageRow) {
+      return {
+        event: {
+          type: "agent_error",
+          created: Date.now(),
+          configurationId: message.configuration.sId,
+          messageId: message.sId,
+          error: {
+            code: "not_found",
+            message: "AgentMessage row not found",
+          },
+        },
+      };
+    }
+    const m = await Message.create(
+      {
+        sId: messageRow.sId,
+        rank: messageRow.rank,
+        conversationId: conversation.id,
+        parentId: messageRow.parentId,
+        version: messageRow.version + 1,
+        agentMessageId: (
+          await AgentMessage.create(
+            {
+              status: "created",
+              agentConfigurationId: agentMessageRow.agentConfigurationId,
+            },
+            { transaction: t }
+          )
+        ).id,
+      },
+      {
+        transaction: t,
+      }
+    );
+    const agentMessage: AgentMessageType = {
+      id: m.id,
+      sId: m.sId,
+      type: "agent_message",
+      visibility: m.visibility,
+      version: m.version,
+      parentMessageId: message.parentMessageId,
+      status: "created",
+      action: null,
+      content: null,
+      feedbacks: [],
+      error: null,
+      configuration: message.configuration,
+    };
+    return {
+      event: {
+        type: "agent_message_new",
+        created: Date.now(),
+        configurationId: agentMessage.configuration.sId,
+        messageId: agentMessage.sId,
+        message: agentMessage,
+      },
+      agentMessageRow,
+    };
+  });
+
+  if (event.type === "agent_error") {
+    return event;
+  }
+  // so that typescript is fine with streamRunAgentEvents using agentMessageRow
```

Comment:
comments should be as much as possible sentences. Started by upper case and finished by a `.` :pray: